asset/file_reader/index.js

```diff
@@ -0,0 +1,103 @@
+'use strict';
+
+const Promise = require('bluebird');
+const Queue = require('@terascope/queue');
+const { getChunk, getOffsets } = require('@terascope/chunked-file-reader');
+const fse = require('fs-extra');
+
+function newSlicer(context, executionContext, retryData, logger) {
+    const opConfig = executionContext.config.operations[0];
+    const queue = new Queue();
+
+    async function processFile(filePath) {
+        const stat = await fse.stat(filePath);
+        const total = stat.size;
+        getOffsets(opConfig.size, total, opConfig.delimiter).forEach((offset) => {
+            offset.path = filePath;
+            queue.enqueue(offset);
+        });
+    }
+
+    async function getFilePaths(filePath) {
+        const dirContents = await fse.readdir(filePath);
+        return Promise.map(dirContents, async (file) => {
+            const fullPath = `${filePath}/${file}`;
+            const stats = await fse.stat(fullPath);
+            if (stats.isFile()) {
+                return processFile(fullPath);
+            }
+            if (stats.isDirectory()) {
+                return getFilePaths(fullPath);
+            }
+            logger.error(`${file} is not a file or directory!!`);
+            return true;
+        })
```

Comment:
Depending on our answer letting unhandled errors go & fail the job, the `fse` & `stats` calls have potential to throw, so we may want to catch & handle a certain way.

Reply:
The short-term plan will just be to let errors bubble up.